phoenix-core-client/src/main/java/org/apache/phoenix/jdbc/PhoenixPrefetchedResultSet.java

```diff
@@ -42,7 +42,8 @@ public PhoenixPrefetchedResultSet(RowProjector rowProjector,
 
   @Override
   protected Tuple getCurrentRowImpl() {
-    return prefetchedRows.get(prefetchedRowsIndex++);
+    return prefetchedRows.size() > prefetchedRowsIndex ? prefetchedRows.get(prefetchedRowsIndex++)
+            : null;
```

Comment:
Nice! This was in my backlog, missed it for sometime.

Reply:
Thanks! I also [added](https://github.com/apache/phoenix/pull/2176/files#diff-064ec4a242ed8c46fdd6c4597fa03cf2f7aae5366c97aba1d87c3ae8b15f3017R503) an explicit next() to catch this.